ui/app/AppLayouts/Wallet/popups/swap/SwapApproveCapModal.qml

```diff
@@ -0,0 +1,240 @@
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Utils 0.1 as SQUtils
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+
+import AppLayouts.Wallet 1.0
+import AppLayouts.Wallet.panels 1.0
+import AppLayouts.Wallet.popups 1.0
+
+import shared.controls 1.0
+import utils 1.0
+
+SignTransactionModalBase {
+    id: root
+
+    required property string fromTokenSymbol
+    required property string fromTokenAmount
+    required property string fromTokenContractAddress
+
+    required property string accountName
+    required property string accountAddress
+    required property string accountEmoji
+    required property color accountColor
+    required property string accountBalanceAmount
+
+    required property string networkShortName // e.g. "oeth"
+    required property string networkName // e.g. "Optimism"
+    required property string networkIconPath // e.g. `Style.svg("network/Network=Optimism")`
+    required property string networkBlockExplorerUrl
+
+    required property string currentCurrency
+    required property string fiatFees
+    required property string cryptoFees
+    // need to check how this is done in new router, right now it is Enum type
+    required property int estimatedTime // Constants.TransactionEstimatedTime.XXX enum
+
+    property string serviceProviderName: "Paraswap"
+    property string serviceProviderURL: Constants.swap.paraswapUrl // TODO https://github.com/status-im/status-desktop/issues/15329
+    property string serviceProviderContractAddress: "0x1bD435F3C054b6e901B7b108a0ab7617C808677b"
+
+    title: qsTr("Approve spending cap")
+    subtitle: serviceProviderURL
+
+    gradientColor: Utils.setColorAlpha(root.accountColor, 0.05) // 5% of wallet color
+    fromImageSmartIdenticon.asset.name: "filled-account"
+    fromImageSmartIdenticon.asset.emoji: root.accountEmoji
+    fromImageSmartIdenticon.asset.color: root.accountColor
+    fromImageSmartIdenticon.asset.isLetterIdenticon: !!root.accountEmoji
+    toImageSource: Constants.tokenIcon(root.fromTokenSymbol)
+
+    //: e.g. "Set 100 DAI spending cap in <account name> for <service> on <network name>"
+    headerMainText: qsTr("Set %1 %2 spending cap in %3 for %4 on %5").arg(formatBigNumber(root.fromTokenAmount)).arg(root.fromTokenSymbol)
+        .arg(root.accountName).arg(root.serviceProviderURL).arg(root.networkName)
+    headerSubTextLayout: [
+        StatusBaseText {
+            Layout.fillWidth: true
+            horizontalAlignment: Qt.AlignHCenter
+            wrapMode: Text.WrapAtWordBoundaryOrAnywhere
+            font.pixelSize: Style.current.additionalTextSize
+            text: qsTr("The smart contract specified will be able to spend up to %1 %2 of your current or future balance.").arg(formatBigNumber(root.fromTokenAmount)).arg(root.fromTokenSymbol)
+        }
+    ]
+
+    headerIconComponent: StatusSmartIdenticon {
+        asset.name: Style.png("swap/paraswap") // FIXME svg
```

Comment:
Why not downloading the SVG from Figma? 😄

Reply:
Yeah, we might want to as some point, the current PNG we have is just disgustingly pixelated :)